src/lib/utils/thorchain/memo/assertAndProcessMemo.ts

```diff
@@ -82,13 +82,33 @@ export const assertAndProcessMemo = (memo: string): string => {
     case '=':
     case 's': {
       // SWAP:ASSET:DESTADDR:LIM/INTERVAL/QUANTITY:AFFILIATE:FEE
-      const [_action, asset, destAddr, limit, , fee] = memo.split(':')
+      const [
+        _action,
+        asset,
+        destAddr,
+        limit,
+        ,
+        fee,
+        aggregatorLastTwoChars,
+        finalAssetAddressShortened,
+        finalAssetAmountOut,
+      ] = memo.split(':')
 
       assertMemoHasAsset(asset, memo)
       assertMemoHasDestAddr(destAddr, memo)
       assertIsValidLimit(limit, memo)
 
-      return `${_action}:${asset}:${destAddr}:${limit}:${THORCHAIN_AFFILIATE_NAME}:${fee || 0}`
+      const swapOutParts = (() => {
+        if (aggregatorLastTwoChars && finalAssetAddressShortened && finalAssetAmountOut) {
```

Comment:
Probably want to use this check instead - if one is defined but not all, it *is* a SwapOut memo, although one we constructed wrongly, which the suggested assertions under will catch.
```suggestion
        if (aggregatorLastTwoChars || finalAssetAddressShortened || finalAssetAmountOut) {
```

Reply:
Very nice catch

Reply:
Done in 03978b18e16